### DISCUSSION TITLE
MulitSelectDropDown - option to make `selected` not a list when in single mode

### DISCUSSION THREAD

**Discussion by rhurlbatt** [anvilistas/anvil-extras#247] [Ideas]
If `multiple` is not selected, it would be nice to be able to use this component as a regular dropdown. ie accept a single element and return a single element. This would have to be by option so that it doesn't mess with existing usage but would make data bindings a whole lot easier.

Unless I am missing some thing simple here so I don't have to always index the first element?

**Comment by s-cork** [comment]
Sure - this makes sense to me. I guess I'd thought that it didn't really make sense use the MultiSelectDropDown without `multiple` ticked. But since `multiple` was an easy api addition I added it.

Another option - which might be the one to go with.
Include `selected_value` just like anvil's dropdown api. 
This way it's backwards compatible. And would work with both `multiple` or not `multiple`

It can just return the first element from `selected`

**Reply by rhurlbatt** [reply]
One of the best things about this component in single mode is the ability to filter the list! I am currently replacing all my standard dropdowns with this.

> Include selected_value just like anvil's dropdown api.

Agreed, this makes a lot more sense.
